WebInstall/explainer_cross_domain.md

```diff
@@ -0,0 +1,259 @@
+/* tries to install a cross-origin web app */
+
+const installApp = async (manifest_id) => {
+    if ('install' in navigator === false) return; // api not supported
+    try {
+            await navigator.install(manifest_id);
+    } catch(err) {
+        switch(err.message){
+            case 'AbortError':
+                /* Operation was aborted*/
+                break;
+           
+        }
+    }
+};
+```
+
+  ![Install flow from an app repository](./apprepositoryinstallation.png) 
+
+## Proposed Solution
+
+### The `navigator.install` method
+
+To install a web app, a web site would use the promise-based method `navigator.install([<manifest_id>[, <params>]]);`. This method will:
+
+* Resolve when an installation was completed.
+    * The success value will be an object that contains:
+     	*  `mode`: string with the surface-hint where the app was installed.
+* Be rejected if the prompt is not shown or if the app installation did not complete. It'll reject with a [`DOMException`](https://developer.mozilla.org/en-US/docs/Web/API/DOMException) value of:
+    * `AbortError`: The installation (prompt) was closed/cancelled.
+
+
+![Promises resolve/reject flow](./installPromises.png) 
+
+
+#### **Signatures of the `install` method (cross-origin)**
+The cross-origin part of the Web Install API consists of the extension to the navigator interface with an `install` method. This method receives:
+* `manifest_id`: decalres the specific application to be installed. If the content to be installed is not content with a manifest, a `install_url` parameter can be specified.
+* optional [parameters](#parameters).
+
+Unless the UA decides to [gate this functionality behind installation](#gating-capability-behind-installation), the behaviour between calling the `install` method on a tab or on an installed application should not differ. The install method can be used in two different ways.
+
+1. `navigator.install(<manifest_id>)`: The method receives a parameter which is a [manifest id](https://w3c.github.io/manifest/#id-member) to a web app to install. This will prompt for installation of the app if the requesting origin has installation permissions (see [security section](#integration-with-the-permissions-api)). This is the most common use of the API.
+
+2. `navigator.install(<manifest_id>, <params>)`: This signature of the method includes the optional parameters. These parameters alter how the app is installed and are defined in an object. More information about the parameters is found in the [Parameters](#parameters) subsection of this specification.
+
+#### **Parameters**
+
+The `navigator.install` call can receive an object with a set of parameters that specify different installation behaviours for the app. It is also a way to future-proof the API if additional data were required with the call.
+* **install_url**: Provides a url to install if the specified `manifest_id` can't be found. The UA can attempt to install this origin instead. 
```

Comment:
The way that this is written implies that it will still try to install from the location indicated in `manifest_id` _first_.  From discussions, I thought that the existence of `install_url` meant that it would instead go directly to trying to install from `install_url`, and merely confirm that the manifest_id identified in the manifest at that location matched the one provided as the first parameter.